### PR TITLE
[CBS] Refactor eventing domain

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

```
query Q {
  triggers(namespace: "develop") {
    name
    namespace
    status {
    reason
    status
    }
    spec {
      broker
      filter
      subscriber {
        ref{
          apiVersion
          kind
          name
          namespace
        }
        uri
      }
    }
  }
}
query QRef {
  triggers(
    namespace: "develop"
    subscriber: {
      ref: {
        apiVersion: "serving.knative.dev/v1"
        kind: "Service"
        name: "tt2"
        namespace: "develop"
      }
    }
  ) {
    name
    spec {
      subscriber {
        ref {
          apiVersion
          kind
          name
          namespace
        }
        uri
      }
    }
  }
}
query QUri {
  triggers(
    namespace: "develop"
    subscriber: {
      uri: "grzesiu"
    }
  ) {
    name
    spec {
      subscriber {
        ref {
          apiVersion
          kind
          name
          namespace
        }
        uri
      }
    }
  }
}
mutation C1 {
  createTrigger(
    namespace: "develop"
    trigger: {
      namespace: "develop"
      broker: "default"
      filterAttributes: {
        eventtypeversion: "v1"
        source: "develop"
        type: "tets8"
      }
      subscriber: {
        ref: {
          apiVersion: "serving.knative.dev/v1"
          kind: "Service"
          name: "abc"
          namespace: "develop"
        }
        uri: "aaa.bc"
      }
    }
  ) {
    name
    namespace
    status {
    reason
    status
    }
    spec {
      broker
      filter
      subscriber {
        ref{
          apiVersion
          kind
          name
          namespace
        }
        uri
      }
    }
  }
}
mutation CN {
  createManyTriggers(
    namespace: "develop"
    triggers: [
      {
        namespace: "develop"
        broker: "default"
        filterAttributes: {
          eventtypeversion: "v1"
          source: "develop"
          type: "tets5"
        }
        subscriber: {
          uri: null
          ref: {
            apiVersion: "v1"
            kind: "Service"
            name: "n1"
            namespace: "develop"
          }
        }
      }
      {
        name: "n2"
        namespace: "develop"
        broker: "default"
        filterAttributes: {
          eventtypeversion: "v1"
          source: "develop"
          type: "tets6"
        }
        subscriber: {
          uri: null
          ref: {
            apiVersion: "v1"
            kind: "Service"
            name: "n2"
            namespace: "develop"
          }
        }
      }
    ]
  ) {
    name
  }
}
mutation D1 {
  deleteTrigger(namespace: "develop", triggerName: "tiger") {
    name
  }
}
mutation DN {
  deleteManyTriggers(namespace: "develop", triggerNames: ["qqq", "qqq1", "qqq2", "qqq3", "qqq4"]) {
    name
  }
}
subscription S {
  triggerEvent(namespace: "develop") {
    type
    trigger {
      name
    }
  }
}
subscription SRef {
  triggerEvent(
    namespace: "develop"
    subscriber: {
      ref: {
        apiVersion: "v1"
        kind: "Service"
        name: "n"
        namespace: "develop"
      }
    }
  ) {
    type
    trigger {
      name
    }
  }
}
subscription SUri {
  triggerEvent(
    namespace: "develop"
    subscriber: {
      uri: "aaa.bc"
    }
  ) {
    type
    trigger {
      name
    }
  }
}
```

**Related issue(s)**
[Console](https://github.com/kyma-project/console/pull/1889)

### CODE REVIEW

File: components/console-backend-service/internal/domain/eventing/resolver.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	return e
+}
+
+func (r *Resolver) TriggersQuery(ctx context.Context, namespace string, subscriber *duckv1.Destination) ([]*v1alpha1.Trigger, error) {
+	items := TriggerList{}
+	var err error
+
+	if subscriber == nil {
+		err = r.Service().ListInNamespace(namespace, &items)
+	} else if subscriber.Ref != nil {
```
Comment (piotrmiskiewicz) [inline components/console-backend-service/internal/domain/eventing/resolver.go:28]: which index should be used when Ref and URI is set? maybe it is not a case, I don't know the trigger domain and we can skip it

Comment (Wawrzyn321) [reply components/console-backend-service/internal/domain/eventing/resolver.go:28]: Me neither, but we could always introduce a combined index for both ref and URI, just like in hostname+service index in apigateway.

File: components/console-backend-service/internal/domain/eventing/service.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+var triggersGroupVersionResource = schema.GroupVersionResource{
+	Version:  v1alpha1.SchemeGroupVersion.Version,
+	Group:    v1alpha1.SchemeGroupVersion.Group,
+	Resource: "triggers",
+}
+
+var triggersSubscriberRefIndex = "subscriberRef"
+
+func triggersSubscriberRefIndexKey(namespace string, subscriberRef *v1.KReference) string {
```
Comment (parostatkiem) [inline components/console-backend-service/internal/domain/eventing/service.go:27]: how about `getSubscriberRefIndexKey` ?

File: components/console-backend-service/internal/domain/eventing/service.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		namespace,
+		subscriberRef.APIVersion,
+		subscriberRef.Kind,
+		subscriberRef.Name,
+		subscriberRef.Namespace)
+}
+
+var triggersSubscriberURIIndex = "subscriberURI"
+
+func triggersSubscriberRefURIKey(namespace string, uri *apis.URL) string {
```
Comment (parostatkiem) [inline components/console-backend-service/internal/domain/eventing/service.go:41]: Same as above ^
I think the word `get` would clarify this is a function, not a variable.

**Pull request merged**